test/kube_checks.go

```diff
@@ -104,12 +104,32 @@ func WaitForAllPodsRunning(client *KubeClient, namespace string) error {
 	return WaitForPodListState(client, PodsRunning, "PodsAreRunning", namespace)
 }
 
+// WaitForPodRunning waits for the given pod to be in running state
+func WaitForPodRunning(client *KubeClient, name string, namespace string) error {
+	p := client.Kube.CoreV1().Pods(namespace)
+	return wait.PollImmediate(interval, podTimeout, func() (bool, error) {
+		p, err := p.Get(name, metav1.GetOptions{})
+		if err != nil {
+			return true, err
+		}
+		return PodRunning(p), nil
+	})
+}
+
 // PodsRunning will check the status conditions of the pod list and return true all pods are Running
 func PodsRunning(podList *corev1.PodList) (bool, error) {
 	for _, pod := range podList.Items {
-		if pod.Status.Phase != corev1.PodRunning && pod.Status.Phase != corev1.PodSucceeded {
+		if isRunning := PodRunning(&pod); !isRunning {
 			return false, nil
 		}
 	}
 	return true, nil
 }
+
+// PodRunning will check the status conditions of the pod and return true if it's Running
+func PodRunning(pod *corev1.Pod) bool {
+	if pod.Status.Phase != corev1.PodRunning && pod.Status.Phase != corev1.PodSucceeded {
```

Comment:
and now we can inline this into: 
`return pod.Status.Phase == corev1.PodRunning || pod.Status.Phase == corev1.PodSucceeded` :-)

Reply:
Done. Thanks!

/meow